airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py

```diff
@@ -74,18 +74,35 @@ def get_authenticator(config):
 
     def check_connection(self, logger, config) -> Tuple[bool, any]:
         config = self._validate_and_transform(config)
-        authenticator = self.get_authenticator(config)
+
+        config = config.get("credentials") or config
+
         url = f"{PinterestStream.url_base}user_account"
-        auth_headers = {"Accept": "application/json", **authenticator.get_auth_header()}
+        auth_headers = {}
         try:
+            authenticator = any
+            if config.get("auth_method") == "access_token":
```

Comment:
Why we compare with `access_token`. In spec we have only `oauth2.0` auth_method value

Reply:
The contribution was made before https://github.com/airbytehq/airbyte/pull/31876 where Pinterest had two different auth methods.

Reply:
so I believe now it is not relevant? or we need change condition